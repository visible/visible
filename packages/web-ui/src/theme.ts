export interface Theme {
  foreground: {
    normal: string;
    wash: string;
  };
  background: {
    normal: string;
    wash: string;
  };
  highlight: {
    normal: string;
  };
  border: {
    normal: string;
  };
}
