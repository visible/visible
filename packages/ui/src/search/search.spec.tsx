import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '../theme';
import { Search } from '.';

describe('Search', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Search />
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls onSubmit', () => {
    const onSubmit = jest.fn();
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <Search submitLabel="Submit" onSubmit={onSubmit} />
      </ThemeProvider>,
    );

    fireEvent.click(getByText('Submit'));
    expect(onSubmit).toBeCalled();
  });

  it('calls onChange', () => {
    const onChange = jest.fn();
    const { getByPlaceholderText } = render(
      <ThemeProvider theme={theme}>
        <Search placeholder="Type your text" onChange={onChange} />
      </ThemeProvider>,
    );

    fireEvent.change(getByPlaceholderText('Type your text'), {
      target: { value: 'New text' },
    });

    expect(onChange).toBeCalled();
  });
});
