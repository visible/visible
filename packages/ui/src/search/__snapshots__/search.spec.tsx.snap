// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches snapshot 1`] = `
.c4 {
  box-sizing: border-box;
  padding: 8px 14px;
  -webkit-transition: 0.15s ease-in;
  transition: 0.15s ease-in;
  border: none;
  border-radius: 4px;
  font-size: 14px;
  cursor: pointer;
  background-color: #ffaa01;
  box-shadow: 0 1.5px 3px rgba(255,170,1,0.2);
  color: white;
}

.c4:hover.c3:not(:disabled) {
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  opacity: 0.8;
}

.c4:focus {
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  outline: none;
  box-shadow: 0 0 0 3px rgba(255,170,1,0.4);
}

.c4:active {
  opacity: 1 !important;
}

.c4:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c2 {
  box-sizing: border-box;
  padding: 8px 14px;
  -webkit-transition: 0.15s ease-in;
  transition: 0.15s ease-in;
  border: 1px solid #e6e6e6;
  border-radius: 4px;
  font-size: 14px;
}

.c2:focus {
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  border-color: #ffaa01;
  outline: none;
  box-shadow: 0 0 0 3px rgba(255,170,1,0.4);
}

.c2:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .c3 {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.c0 .c1 {
  border-right: none;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

<form
  class="c0"
  role="search"
>
  <input
    class="c1 c2"
    type="text"
    value=""
  />
  <button
    class="c3 c4"
    type="submit"
  />
</form>
`;
